src/Temporalio/Worker/Interceptors/WorkflowInboundInterceptor.cs

```diff
@@ -65,5 +65,20 @@ private protected WorkflowInboundInterceptor()
         /// <param name="input">Input details of the call.</param>
         /// <returns>Query result.</returns>
         public virtual object? HandleQuery(HandleQueryInput input) => Next.HandleQuery(input);
+
+        /// <summary>
+        /// Intercept update validation. This is invoked every time validation is needed regardless
+        /// of whether a validator was defined.
+        /// </summary>
+        /// <param name="input">Input details of the call.</param>
+        public virtual void ValidateUpdate(HandleUpdateInput input) => Next.ValidateUpdate(input);
```

Comment:
In my implementation I only invoke if the validator is defined. I can see doing it always, if you want some blanket way to reject based on... whatever... without defining a validator for every update. I'll change that in my impl.

Reply:
:+1: Yeah I went back and forth here too but I figured "why not"

Reply:
Ok, after team discussion, here is what we want the SDK logic to be due to possible mutation of input and performance:

* If validator present
  * decode args (or reject on fail)
  * call validator (interceptor) w/ those args
* decode args (or reject on fail)
* accept
* call updator (interceptor) w/ those args

Reply:
The logic has been updated to support this approach

Reply:
Sorry to beat a dead horse, but I thought if we have to decode the args we would go through the interceptor

Reply:
because it is a non trivial change in go and java to even know if a validator is present or not without going through the interceptor

Reply:
> Sorry to beat a dead horse
No problem, we need to get this right/consistent. Happy to discuss as much as is necessary.

So double-decoding args has already been settled to prevent validator mutability. We will discuss as a team whether to prevent interceptor call if validator present or not.

Reply:
Ok, after discussion, the logic in the bullets above is what we want. Don't run validate interceptor (and therefore don't decode args for it) if a validator is not present